javaslang/src/main/java/javaslang/$.java

```diff
@@ -33,8 +33,7 @@
     // -- javaslang.collection
 
     // List
-    @Unapply static <T> Tuple2<T, List<T>> List(List.Cons<T> cons) { return Tuple.of(cons.head(), cons.tail()); }
-    @Unapply static <T> Tuple0 List(List.Nil<T> nil) { return Tuple.empty(); }
+    @Unapply static <T> Tuple2<T, List<T>> List(List<T> cons) { return Tuple.of(cons.head(), cons.tail()); }
```

Comment:
Hi @paplorinc, the removed lines should work. These are the patterns that are generated at compile time:

```java
// @Unapply static <T> Tuple2<T, List<T>> List(List.Cons<T> cons)
public static <T, _1 extends T, _2 extends List<T>> Pattern2<List.Cons<T>, _1, _2> List(Pattern<_1, ?> p1, Pattern<_2, ?> p2) {
    return Pattern2.of(List.Cons.class, p1, p2, $::List);
}

// @Unapply static <T> Tuple0 List(List.Nil<T> nil)
public static <T> Pattern0<List.Nil<T>> List() {
    return Pattern0.of(List.Nil.class);
}
```

Some tests are in MatchTest.java. But I will add these (which work with lines deleted above):

```java
@Test
public void shouldDecomposeEmptyList() {
    final List<Integer> list = List.empty();
    final boolean isEmpty = Match(list).of(
            Case(List($(), $()), (x, xs) -> false),
            Case(List(), true)
    );
    assertThat(isEmpty).isTrue();
}

@Test
public void shouldDecomposeNonEmptyList() {
    final List<Integer> list = List.of(1);
    final boolean isNotEmpty = Match(list).of(
            Case(List($(), $()), (x, xs) -> true),
            Case(List(), false)
    );
    assertThat(isNotEmpty).isTrue();
}
```

Could you please explain your observation?